README.md

```diff
@@ -81,69 +81,116 @@ Summary of one-time project setup:
+4. Read through the test failure, and understand why the failure is happening. Confirm your findings with a classmate. 
+    - If it is a test you wrote, consider whether you are actually testing what you intend to test. Does the test need modification?
+
+5. Make a plan to implement code to pass the test.
 
-5. Make a plan to fix the test failure.
+6. Write code to pass the test (GREEN).
 
-6. Write code to fix the test failure.
+7. Re-run the test(s).
 
-7. Re-run the tests.
+8. Repeat steps 3-7 until that test passes!
 
-8. Repeat steps 5-7 until that test passes!
+9. Consider looking for opportunities to improve your code (REFACTOR)
 
-9. Repeats steps 4-8 until you have finished all tests in the file.
+10. Repeats steps 2-8 until you have finished all tests in the file.
 
-10. Begin using the test file of the next wave!
+11. Begin using the test file of the next wave!
 
-11. When you are finished working for the day, deactivate your environment with deactivate or closing the Terminal tab/window
+12. When you are finished working for the day, deactivate your environment with deactivate or closing the Terminal tab/window
 
 ```bash
 $ deactivate
 ```
+Finally, at submission time, **no matter where you are**, submit the project via Learn.
+
+This will let us give feedback on what you've finished so that you can be better prepared for the next project.
+
+## Integration Tests
+
+We provided integration tests for this project. Integration tests verify that individual units of a piece of software are working together correctly.
+
+The integration tests for this project are in the directory `tests/integration_tests`, and the tests have the decorator `@pytest.mark.integration_test` provided by the `pytest-integration` package. Marking these tests as integration tests makes them run after the unit tests. The isolated nature and specficity of *unit tests* make them a much better tool for debugging. Hence, we want to run the unit tests before the integration tests.
+
+The integration tests provided in this project do not cover all the test cases verified by the unit tests. They verify that the functions work together when invoked correctly. We could classify them as the subset of Integration Tests called *Smoke Tests*, tests that check basic functionality of the application.
+
+For more on different types of software testing, [here is a helpful resource](https://www.atlassian.com/continuous-delivery/software-testing/types-of-software-testing). 
+
+## Code Coverage
+
+It is good practice to check code coverage. Code coverage is a term that is used to describe how much application code is executed when a particular test suite is run. A test suite with a high percentage of coverage is likely to be testing more throughly and have fewer bugs. A code coverage tool can partner with our testing suite to give us a report illustrating the coverage of our tests.
```

Comment:
I was thinking it might be helpful to define code coverage first and provide a little more "why" in the good practice sentence. 

```suggestion
Code coverage is a term used to describe how much application code is executed when a particular test suite is run. It is a good practice to check our code coverage, to understand how much of our code is exercised by tests vs how much is still untested. A test suite with a high percentage of coverage is likely to be testing more throughly and have fewer bugs. A code coverage tool can partner with our testing suite to give us a report illustrating the coverage of our tests.
```